### PR TITLE
New package: gnome-console-42.2

### PR DESCRIPTION
**Pull request by oreo639**
#### Testing the changes
- I tested the changes in this PR: **YES**

https://github.com/void-linux/void-packages/pull/36311

### CODE REVIEW

**Comment by paper42** [comment]
afaik GNOME didn't replace gnome-terminal with gnome-console with GNOME 42, so this is separate from #36311 and should be marked as a draft because this packages a beta version.

**Comment by oreo639** [comment]
> afaik GNOME didn't replace gnome-terminal with gnome-console with GNOME 42

Yes, I just made this separate since it doesn't depend on anything from there.

**Comment by oreo639** [comment]
Oh, sorry I misread your comment, they did replace gnome-terminal with gnome-console for GNOME 42:
https://gitlab.gnome.org/GNOME/gnome-build-meta/-/merge_requests/1404

**Comment by oreo639** [comment]
Should this become apart of the gnome 42 PR (and added to gnome-core) or no?

File: srcpkgs/gnome-console/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'gnome-console'
+pkgname=gnome-console
+version=42.beta
+revision=1
+build_style=meson
+configure_args="-Dnautilus=enabled"
+hostmakedepends="pkg-config desktop-file-utils appstream-glib gettext
```
Comment (paper42) [inline srcpkgs/gnome-console/template:7]:
```suggestion
hostmakedepends="pkg-config desktop-file-utils gettext
```

**Comment by paper42** [comment]
> Should this become apart of the gnome 42 PR (and added to gnome-core) or no? (the vte update is needed by gnome-terminal anyway)

If they replaced gnome-terminal with kgx, we should do the same.

**Comment by oreo639** [comment]
Reopened per https://github.com/void-linux/void-packages/pull/36311#issuecomment-1093905497

**Comment by oreo639** [comment]
Reopened per https://github.com/void-linux/void-packages/pull/36311#issuecomment-1093905497

File: srcpkgs/glib/patches/gnome-console-support.patch [hunk 2]
```diff
@@ -0,0 +1,62 @@
+Currently glib hardcodes a list of terminals and that list will not be expanding.
+The hope being that it will eventually be replaced with some kind of api.
+Until then, we might as well add gnome-console/kgx to the list
+
+See:
+https://gitlab.gnome.org/GNOME/gnome-build-meta/-/merge_requests/1404#note_1331262
+
```
Comment (paper42) [inline srcpkgs/glib/patches/gnome-console-support.patch:7]: please add the source of this patch

Comment (oreo639) [reply srcpkgs/glib/patches/gnome-console-support.patch]: Thank you.

File: srcpkgs/gnome-console/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'gnome-console'
+pkgname=gnome-console
+version=42.2
+revision=1
+build_style=meson
+configure_args="-Dnautilus=enabled"
+hostmakedepends="pkg-config desktop-file-utils gettext sassc
+ gobject-introspection glib-devel gtk+3-devel"
```
Comment (paper42) [inline srcpkgs/gnome-console/template:8]:
```suggestion
 gobject-introspection glib-devel gtk-update-icon-cache "
```

File: srcpkgs/gnome-console/template [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+makedepends="libhandy1-devel vte3-devel libgtop-devel nautilus-devel
+ gsettings-desktop-schemas-devel"
```
Comment (paper42) [inline srcpkgs/gnome-console/template:10]:
```suggestion
 gsettings-desktop-schemas-devel pcre2-devel"
```

**Pull request merged**